### PR TITLE
[content-init] Ensure .gitpod path is available

### LINKED ISSUE #7694: Error : Connecting to workspace logs
### Bug description

* Opening the following repository [Link](https://gitpod.io/#https://github.com/32leaves/gitpod-hello-ui-demo) throws error (which used to work before like may be half a year ago or so):-
  ```sh
  Connecting to workspace logs ... 
  Error: cannot initialize workspace: cannot initialize workspace: content initializer failed
  ```
  image

* Changing filename from `.gitpod` to `.gitpod.yml` also didnt help
* The following repository fork [Link](https://gitpod.io/#https://github.com/hpssjellis/gitpod-hello-ui-demo) atleast goes past the `Connecting to workspace logs .. ` and builds
* Can someone tell whats the difference thats causing one to build and other to not (as both seem like valid .yml files) ?

### Steps to reproduce

* Open the following link via browser:-
  * https://gitpod.io/#https://github.com/32leaves/gitpod-hello-ui-demo
* Following error shows up:-
  ```sh
  Connecting to workspace logs ... 
  Error: cannot initialize workspace: cannot initialize workspace: content initializer failed
    ```

### Workspace affected

_No response_

### Expected behavior

* Open the following link via browser:-
  * https://gitpod.io/#https://github.com/hpssjellis/gitpod-hello-ui-demo
* Logs should go past `Connecting to workspace logs..` :-
  ```sh
  Connecting to workspace logs ... 
  Connecting to log outputs ... 
  ....
  ```

### Example repository

https://github.com/32leaves/gitpod-hello-ui-demo

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by csweichel**
## Description
When repositories contain a `.gitpod` file this can cause the content initializer to fail because we use `/workspace/.gitpod` internally. This change introduces special handling such cases where we try to `git mv` or `mv` the `.gitpod` file to `.gitpod.yaml`. If `.gitpod.yaml` already exists, we delete `.gitpod`.

## Related Issue(s)

Fixes #7694

## How to test
Open https://cw-fix-7694.staging.gitpod-dev.com/#https://github.com/csweichel/gitpod-hello-ui-demo
If the build starts (might not complete due to other issues), we're good. Compare with prod where the build fails to start due to a content init error.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#7694*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Furisto — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 19fffa7b50f5ea26c32bd20de38c2f71cb6b870f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7749#pullrequestreview-859386674" title="Approved">Furisto</a>*

Associated issue: *#7694*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)~~ [Furisto]
- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [Furisto]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
